pyspecProcScripts/calc_error.py

```diff
@@ -0,0 +1,131 @@
+import pyspecdata as psd
+from .simple_functions import select_pathway
+import numpy as np
+
+
+# {{{ Functions to calculate var and mean of nan-masked data
+#     Pulled from time_domain_noise.py example
+def masked_mean_multi(x, axis=None):
+    "Calculate the mean of nan-masked data on a 1D axis"
+    assert axis is not None
+
+    def masked_mean(x):
+        return np.mean(x[np.isfinite(x)])
+
+    return np.apply_along_axis(masked_mean, axis, x)
+
+
+def masked_var_multi(x, var_has_imag=False, axis=None):
+    "Calculates the variance along a 1D axis"
+    assert axis is not None
+
+    def masked_var(x):
+        if var_has_imag:  # take average of variance along real and imag
+            return np.var(x[np.isfinite(x)], ddov=1) / 2
+        else:
+            return np.var(x[np.isfinite(x)], ddof=1)
+
+    return np.apply_along_axis(masked_var, axis, x)
+
+
+# }}}
+def calc_error(
+    s,
+    frq_slice,
+    signal_pathway,
+    error_path,
+    direct="t2",
+    indirect="nScans",
+    use_mask=False,
```

Comment:
the default is to use the error_path as we've been using it but if True this is going to use the nan mask

Reply:
see comments about using `excluded_pathways` instead